src/main/bootstrap/mod.rs

```diff
@@ -430,6 +440,58 @@ mod tests {
         }
     }
 
+    mod update_peers {
+        use super::*;
+
+        #[test]
+        fn it_filters_out_duplicates() {
+            let mut peers: Vec<PeerInfo> = (0..10)
+                .map(|i| peer_info_with_rand_key(ipv4_addr(1, 2, 3, 4, i)))
+                .collect();
+
+            let (tx, rx) = mpsc::channel();
+
+            let duplicates: Vec<PeerInfo> = peers.iter().take(5).map(|peer| peer.clone()).collect();
+            tx.send(duplicates).unwrap();
+
+            update_peers(&mut peers, rx);
+
+            let peers_len = peers.len();
+            let mut peers_set = HashSet::new();
```

Comment:
What about simple `let peers_set: HashSet<PeerInfo> = peers.into_iter().cloned().collect();`?

Reply:
I wanted to be able to assert if any of the `.insert()` calls would fail, which means the list contains duplicates, but I guess only checking if the length of the set and the vector are the same will be enough, so I will change that too.